datafusion/expr/src/logical_plan/extension.rs

```diff
@@ -94,6 +94,9 @@ pub trait UserDefinedLogicalNode: fmt::Debug + Send + Sync {
     /// Update the hash `state` with this node requirements from
     /// [`Hash`].
     ///
+    /// Note: consider using [`UserDefinedLogicalNodeCore`] instead of
```

Comment:
That's a great hint for people doing the upgrade. 👏

Next time I write docs I'll take a while to take the perspective of a person doing an update. 💡